src/raven/client.clj

```diff
@@ -188,16 +185,26 @@
                 (.doFinal (.getBytes (format "%s %s" ts payload))))]
     (reduce str (for [b bs] (format "%02x" b)))))
 
+(defn get-http-client
+  "Get a http client given a context.
+    
+    We expect callers to pass the http client in the context object at the
+    :http key."
+  [context]
+  (cond
+    (contains? context :http)   (:http context)
+    :else                       (http/build-client {})))
```

Comment:
`(or (:http context) (http/build-client {}))` feels similar

Reply:
Ohhh thanks. That is indeed a better way to solve this :)